controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -80,10 +80,10 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	if instance.Status.Conditions == nil {
 		reason := status.ReconcileInit
 		message := "Initializing DSCInitialization resource"
-		status.SetProgressingCondition(&instance.Status.Conditions, reason, message)
-
-		instance.Status.Phase = status.PhaseProgressing
-		err = r.Client.Status().Update(ctx, instance)
+		instance, err = r.updateStatus(instance, func(saved *dsci.DSCInitialization) {
+			status.SetProgressingCondition(&saved.Status.Conditions, reason, message)
+			saved.Status.Phase = status.PhaseProgressing
+		})
 		if err != nil {
 			r.Log.Error(err, "Failed to add conditions to status of DSCInitialization resource.", "DSCInitialization", req.Namespace, "Request.Name", req.Name)
```

Comment:
should here to use `r.reportError() `?

Reply:
We don't have a reportError() method for the DSCInitializationReconciler.